src/components/AQIGauge/GaugeSVG.tsx

```diff
@@ -0,0 +1,190 @@
+      percent = aqi / max;
+    } else if (aqi > max) {
+      percent = max;
+    }
+    return percent;
+  };
+
+  // Convert from percent to angle
+  const convertToAngle = scaleLinear()
+    .domain([0, 1]) /* eslint-disable-line no-magic-numbers */
+    .range([startAngle, endAngle])
+    .clamp(true);
+
+  const arcGenerator = arc(); // Used to make arcs
+
+  // Background Arcs
+  const goodArcAngle = convertToAngle(convertToPercent(good, min, max));
+  const goodArc =
+    arcGenerator({
+      innerRadius: innerRadius,
+      outerRadius: outerRadius,
+      startAngle: startAngle,
+      endAngle: goodArcAngle,
+    }) ?? '';
+  const moderateArcAngle = convertToAngle(convertToPercent(moderate, min, max));
+  const moderateArc =
+    arcGenerator({
+      innerRadius: innerRadius,
+      outerRadius: outerRadius,
+      startAngle: goodArcAngle,
+      endAngle: moderateArcAngle,
+    }) ?? '';
+  const sensitiveArcAngle = convertToAngle(
+    convertToPercent(sensitive, min, max)
+  );
+  const sensitiveArc =
+    arcGenerator({
+      innerRadius: innerRadius,
+      outerRadius: outerRadius,
+      startAngle: moderateArcAngle,
+      endAngle: sensitiveArcAngle,
+    }) ?? '';
+  const unhealthyArcAngle = convertToAngle(
+    convertToPercent(unhealthy, min, max)
+  );
+  const unhealthyArc =
+    arcGenerator({
+      innerRadius: innerRadius,
+      outerRadius: outerRadius,
+      startAngle: sensitiveArcAngle,
+      endAngle: unhealthyArcAngle,
+    }) ?? '';
+  const veryUnhealthyArcAngle = convertToAngle(
+    convertToPercent(veryUnhealthy, min, max)
+  );
+  const veryUnhealthyArc =
+    arcGenerator({
+      innerRadius: innerRadius,
+      outerRadius: outerRadius,
+      startAngle: unhealthyArcAngle,
+      endAngle: veryUnhealthyArcAngle,
+    }) ?? '';
+
+  // Filled arc to overlay the background arcs
+  const filledAngle = convertToAngle(convertToPercent(aqi, min, max));
+  const filledArc =
+    arcGenerator({
+      innerRadius: innerRadius,
+      outerRadius: outerRadius,
+      startAngle: startAngle,
+      endAngle: filledAngle,
+    }) ?? '';
+
+  // Assign appropriate color
+  const assignColor = (aqi: number) => {
+    let arcColor = 'white';
+    if (aqi <= good) {
+      arcColor = '#08E400';
+    } else if (aqi <= moderate) {
+      arcColor = '#FEFF00';
+    } else if (aqi <= sensitive) {
+      arcColor = '#FF7E02';
+    } else if (aqi <= unhealthy) {
+      arcColor = '#FF0202';
+    } else if (aqi <= veryUnhealthy) {
+      arcColor = '#8F3F97';
+    } else {
+      // Hazardous
+      arcColor = '#7E0224';
+    }
+    return arcColor;
+  };
+
+  const needleColor = '#636360';
+
+  return (
+    <div className="svg">
+      <svg height="150" width="300" viewBox={' -1.05 -1.05 2.1 1.1'}>
+        <path
+          d={goodArc}
+          fill={assignColor(good)}
+          stroke="black"
+          strokeWidth={0.002}
+          fillOpacity="0.3"
+        />
+        <path
```

Comment:
I think we might be able to use a for loop of some sort here, given that we do the same calculations (and here, the same path) repeatedly, maybe iterate through a loop of levels, like `for level in [good, moderate, ...]`

Reply:
I tried to do this where I used eval() so that I only needed to maintain one list, but got eslint warnings that it wasn't safe. A google search revealed that eval() can open you up to injection attacks. So I implemented a for loop, but had to maintain multiple lists (for the  colors and for the arcs)